lib/DDG/Spice/OpenNIC.pm

```diff
@@ -0,0 +1,68 @@
+package DDG::Spice::OpenNIC;
+# ABSTRACT: Displays The nearest OpenNIC DNS servers
+
+use DDG::Spice;
+spice is_cached => 1;
+
+# Metadata
+name "OpenNIC";
+source "OpenNIC API";
+description "Search for OpenNIC DNS using some criteras";
+primary_example_queries "opennic", "neutral dns" , "censor-free dns";
+secondary_example_queries "opennic 42.42.42.42", "opennic ipv6", "opennic ipv6 42.42.42.42";
+category "cheat_sheets";
+topics "sysadmin";
+
+# Code info
+code_url "https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/lib/DDG/Spice/OpenNIC.pm";
+attribution github  => ["cylgom", "cylgom"], # it's me
+            web     => ['http://cylgom.net', 'cylgom'],
+            twitter => "cylgom";
+            
+# OpenNIC API endpoint
+    # We prepare a 'spice from' with the place for 4 arguments
+spice from => '([^/]+)/?(?:([^/]+)/?(?:([^/]+)/?(?:([^/]+)/?(?:([^/]+)/?(?:([^/]+)|)|)|)|)|)';
```

Comment:
Are all of these optional? If not, you can drop the `|)`'s. Also it looks like you have 6 captures here?

```perl
spice from => '([^/]+)/([^/]+)/([^/]+)/([^/]+)/([^/]+)'; 
```